OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,349 @@
+        {
+            UpdateCurrentCombination();
+            return _currentCombination != null && Time.realtimeSinceStartup - _timeout[(int)code] < Cooldown;
+        }
+
+        public ModInputHandler(IModLogger logger, IModConsole console, IHarmonyHelper patcher)
+        {
+            _console = console;
+            _logger = logger;
+            patcher.AddPrefix<SingleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisRemovePre));
+            patcher.AddPrefix<DoubleAxisCommand>("ClearBinding", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisRemovePre));
+            patcher.AddPostfix<SingleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.SingleAxisUpdatePost));
+            patcher.AddPostfix<DoubleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisUpdatePost));
+            Instance = this;
+        }
+
+        private long HashFromKeyboard()
+        {
+            long hash = 0;
+            var keysCount = 0;
+            var keys = new int[MaxComboLength];
+            var countdownTrigger = true;
+            for (var code = MinUsefulKey; code < MaxUsefulKey; code++)
+            {
+                if (!(Enum.IsDefined(typeof(KeyCode), (KeyCode)code) && UnityEngine.Input.GetKey((KeyCode)code)))
+                {
+                    continue;
+                }
+                keys[keysCount] = code;
+                keysCount++;
+                if (keysCount > MaxComboLength)
+                {
+                    return (int)RegistrationCode.CombinationTooLong;
+                }
+                hash = hash * MaxUsefulKey + code;
+                if (Time.realtimeSinceStartup - _timeout[code] > Cooldown)
+                {
+                    countdownTrigger = false;
+                }
+            }
+            return countdownTrigger ? -hash : hash;
+        }
+
+        private IModInputCombination CombinationFromKeyboard()
+        {
+            var countdownTrigger = false;
+            var hash = HashFromKeyboard();
+            if (hash < 0)
+            {
+                countdownTrigger = true;
+                hash = -hash;
+            }
+            if (!_comboRegistry.ContainsKey(hash))
+            {
+                return null;
+            }
+
+            var combination = _comboRegistry[hash];
+            if (!(combination == _currentCombination) && countdownTrigger)
+            {
+                return null;
+            }
+
+            if (hash < MaxUsefulKey)
+            {
+                return combination;
+            }
+            while (hash > 0)
+            {
+                _timeout[hash % MaxUsefulKey] = Time.realtimeSinceStartup;
+                hash /= MaxUsefulKey;
+            }
+            return combination;
+        }
+
+        private void UpdateCurrentCombination()
+        {
+            if (_lastCombinationUpdate == Time.frameCount)
```

Comment:
If a frame is skipped, does this check break?

Reply:
frameCount is updated on every new frame and Update is tied to frame render (Unity's order of execution), it's also impossible to overflow that value in reasonable time. Only thing that might actually break it is if someone tries to interact with InputHandler in OnGUI as frameCount might get updated before OnGUI. Although I probably should protect it from being called in FixedUpdate by using Time.inFixedTimeStep

Reply:
Ok. Is there any point in checking with `>=`, just in case?

Reply:
not really, == would work better than >=